File: training-data/perl/bubble_sort.pl
use strict;
use warnings;

sub bubble_sort {
    my @arr = @_;
    for my $i (0 .. $#arr - 1) {
        for my $j (0 .. $#arr - $i - 1) {
            if ($arr[$j] > $arr[$j + 1]) {
                @arr[$j, $j + 1] = @arr[$j + 1, $j];
            }
        }
    }
    return @arr;
}

print join(" ", bubble_sort(5, 2, 9, 1, 5, 6)), "\n";

File: training-data/swift/bubble_sort.swift
import Foundation

func bubbleSort(_ array: [Int]) -> [Int] {
    var arr = array
    let n = arr.count
    for i in 0..<n {
        for j in 0..<(n - i - 1) {
            if arr[j] > arr[j + 1] {
                arr.swapAt(j, j + 1)
            }
        }
    }
    return arr
}

let data = [5, 2, 9, 1, 5, 6]
print(bubbleSort(data))
